### PR TITLE
Issue #5812: Rename all utility classes to xxxxUtil

### PR DESCRIPTION
**Pull request by pbludov**
Issue #5812

The PR is large, but simple.

### CODE REVIEW

**Comment by romani** [comment]
wercker problem is unrelated to merged fix, I restarted it

**Comment by romani** [comment]
code is ok, ones we fix problem with wercker - we can merge.

**Comment by pbludov** [comment]
>Please also run on your local failed wercker command - ./.ci/wercker.sh no-error-xwiki

No errors reported:
```
[INFO] 
[INFO] --- maven-checkstyle-plugin:2.17:check (default-cli) @ xwiki-commons-tool-webjar-handlers ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] XWiki Commons - Parent POM ......................... SUCCESS [  0.961 s]
[INFO] XWiki Commons - Tools - Parent POM ................. SUCCESS [  0.015 s]
[INFO] XWiki Commons - Tools - License Resources .......... SUCCESS [  0.378 s]
[INFO] XWiki Commons - Tools - Enforcers .................. SUCCESS [  0.442 s]
[INFO] XWiki Commons - Tools - Banned Dependencies Enforcer SUCCESS [  0.064 s]
[INFO] XWiki Commons - Tools - Verification Resources ..... SUCCESS [  0.628 s]
[INFO] XWiki Commons - Root POM ........................... SUCCESS [  0.628 s]
[INFO] XWiki Commons - Tools - Test Framework - Parent POM  SUCCESS [  0.367 s]
[INFO] XWiki Commons - Tools - Test Framework - Simple .... SUCCESS [  0.353 s]
[INFO] XWiki Commons - Core - Parent POM .................. SUCCESS [  0.209 s]
[INFO] XWiki Commons - Stability .......................... SUCCESS [  0.538 s]
[INFO] XWiki Commons - Text ............................... SUCCESS [  0.922 s]
[INFO] XWiki Commons - Component - Parent POM ............. SUCCESS [  0.077 s]
[INFO] XWiki Commons - Component - API .................... SUCCESS [  0.896 s]
[INFO] XWiki Commons - Observation - Parent POM ........... SUCCESS [  0.074 s]
[INFO] XWiki Commons - Observation - API .................. SUCCESS [  0.352 s]
[INFO] XWiki Commons - Component - Observation ............ SUCCESS [  0.296 s]
[INFO] XWiki Commons - Component - Default Implementation . SUCCESS [  0.449 s]
[INFO] XWiki Commons - Context ............................ SUCCESS [  0.438 s]
[INFO] XWiki Commons - Configuration - Parent POM ......... SUCCESS [  0.105 s]
[INFO] XWiki Commons - Configuration - API ................ SUCCESS [  0.269 s]
[INFO] XWiki Commons - Tools - Test Framework - Component . SUCCESS [  0.238 s]
[INFO] XWiki Commons - Blame .............................. SUCCESS [  0.208 s]
[INFO] XWiki Commons - Blame - API ........................ SUCCESS [  0.419 s]
[INFO] XWiki Commons - Loggging ........................... SUCCESS [  0.100 s]
[INFO] XWiki Commons - Logging - API ...................... SUCCESS [  1.264 s]
[INFO] XWiki Commons - Diff ............................... SUCCESS [  0.286 s]
[INFO] XWiki Commons - Diff API ........................... SUCCESS [  0.853 s]
[INFO] XWiki Commons - Diff Display API ................... SUCCESS [  0.379 s]
[INFO] XWiki Commons - Script ............................. SUCCESS [  0.309 s]
[INFO] XWiki Commons - Blame - Script Service ............. SUCCESS [  0.326 s]
[INFO] XWiki Commons - Environment - Parent POM ........... SUCCESS [  0.076 s]
[INFO] XWiki Commons - Environment - API .................. SUCCESS [  0.190 s]
[INFO] XWiki Commons - Cache - Parent POM ................. SUCCESS [  0.059 s]
[INFO] XWiki Commons - Cache - API ........................ SUCCESS [  0.345 s]
[INFO] XWiki Commons - Cache - Tests ...................... SUCCESS [  0.762 s]
[INFO] XWiki Commons - Cache - Infinispan ................. SUCCESS [  0.949 s]
[INFO] XWiki Commons - ClassLoader - Parent POM ........... SUCCESS [  0.058 s]
[INFO] XWiki Commons - ClassLoader - API .................. SUCCESS [  0.267 s]
[INFO] XWiki Commons - ClassLoader - Protocols - Parent POM SUCCESS [  0.058 s]
[INFO] XWiki Commons - ClassLoader - Protocols - JAR ...... SUCCESS [  0.237 s]
[INFO] XWiki Commons - Component - Component Archetype .... SUCCESS [  4.203 s]
[INFO] XWiki Commons - Crypto ............................. SUCCESS [  0.122 s]
[INFO] XWiki Commons - Crypto - Common API ................ SUCCESS [  1.013 s]
[INFO] XWiki Commons - Crypto - Cipher API ................ SUCCESS [  0.571 s]
[INFO] XWiki Commons - Crypto - Password API .............. SUCCESS [  1.735 s]
[INFO] XWiki Commons - Crypto - Signer API ................ SUCCESS [  0.340 s]
[INFO] XWiki Commons - Crypto - PKI API ................... SUCCESS [  0.681 s]
[INFO] XWiki Commons - Crypto - Store ..................... SUCCESS [  0.113 s]
[INFO] XWiki Commons - Cryptographic Services - Storage API SUCCESS [  0.289 s]
[INFO] XWiki Commons - Cryptographic Services - Storage Filesystem SUCCESS [  0.340 s]
[INFO] XWiki Commons - Environment - Common ............... SUCCESS [  0.231 s]
[INFO] XWiki Commons - Environment - Standard ............. SUCCESS [  0.301 s]
[INFO] XWiki Commons - Environment - Servlet .............. SUCCESS [  0.243 s]
[INFO] XWiki Commons - Properties ......................... SUCCESS [  0.716 s]
[INFO] XWiki Commons - Observation - Local ................ SUCCESS [  0.791 s]
[INFO] XWiki Commons - Job ................................ SUCCESS [  0.726 s]
[INFO] XWiki Commons - Logging - Logback .................. SUCCESS [  0.231 s]
[INFO] XWiki Commons - Extension - Parent POM ............. SUCCESS [  0.066 s]
[INFO] XWiki Commons - Extension - API .................... SUCCESS [  1.666 s]
[INFO] XWiki Commons - Extension - Maven .................. SUCCESS [  0.455 s]
[INFO] XWiki Commons - Extension - Handler - Parent POM ... SUCCESS [  0.311 s]
[INFO] XWiki Commons - Extension - Handler - JAR .......... SUCCESS [  1.026 s]
[INFO] XWiki Commons - Extension - Repository - Parent POM  SUCCESS [  0.085 s]
[INFO] XWiki Commons - Extension - Repository - HTTP ...... SUCCESS [  1.490 s]
[INFO] XWiki Commons - Extension - Repository - Maven ..... SUCCESS [  1.399 s]
[INFO] XWiki Commons - Repository ......................... SUCCESS [  0.052 s]
[INFO] XWiki Commons - Repository - Model ................. SUCCESS [  0.306 s]
[INFO] XWiki Commons - Repository - API ................... SUCCESS [  0.170 s]
[INFO] XWiki Commons - Extension - Repository - XWiki ..... SUCCESS [  0.480 s]
[INFO] XWiki Commons - Extension - Repository - Maven - Snapshots Default Repository SUCCESS [  0.529 s]
[INFO] XWiki Commons - Filter - Parent POM ................ SUCCESS [  0.052 s]
[INFO] XWiki Commons - Filter - API ....................... SUCCESS [  0.533 s]
[INFO] XWiki Commons - XML ................................ SUCCESS [  0.824 s]
[INFO] XWiki Commons - Filter - XML ....................... SUCCESS [  1.109 s]
[INFO] XWiki Commons - Filter - Test framework ............ SUCCESS [  0.464 s]
[INFO] XWiki Commons - Filter - Events .................... SUCCESS [  0.063 s]
[INFO] XWiki Commons - Filter - Event - Extension ......... SUCCESS [  0.312 s]
[INFO] XWiki Commons - Filter - Streams ................... SUCCESS [  0.055 s]
[INFO] XWiki Commons - Filter - Stream - Generic XML ...... SUCCESS [  0.978 s]
[INFO] XWiki Commons - Groovy ............................. SUCCESS [  0.379 s]
[INFO] XWiki Commons - Management ......................... SUCCESS [  0.161 s]
[INFO] XWiki Commons - Velocity ........................... SUCCESS [  1.168 s]
[INFO] XWiki Commons - Diff Script API .................... SUCCESS [  0.307 s]
[INFO] XWiki Commons - Tools - Extension Plugin ........... SUCCESS [  0.914 s]
[INFO] XWiki Commons - Tools - Spoon Processor ............ SUCCESS [  0.028 s]
[INFO] XWiki Commons - Tools - XAR Tools .................. SUCCESS [  0.007 s]
[INFO] XWiki Commons - Tools - XAR Tools - XAR Plugin ..... SUCCESS [  0.859 s]
[INFO] XWiki Commons - Tools - XAR Tools - XAR Handlers ... SUCCESS [  0.243 s]
[INFO] XWiki Commons - Tools - WEBJAR XAR Handlers ........ SUCCESS [  0.287 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 46.120 s
[INFO] Finished at: 2018-05-27T10:14:07-04:00
[INFO] Final Memory: 278M/9874M
[INFO] ------------------------------------------------------------------------
```

**Comment by romani** [comment]
it is regression on user side after your update:
https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-tools/xwiki-commons-tool-verification-resources/src/main/java/org/xwiki/tool/checkstyle/UnstableAnnotationCheck.java#L35

they use our util class: `import com.puppycrawl.tools.checkstyle.utils.AnnotationUtility;`

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:
3.7.0:compile (default-compile) on project xwiki-commons-tool-verification-resources:
Compilation failure: 
Compilation failure: 
[ERROR] /pipeline/source/.ci-temp/xwiki-commons/xwiki-commons-tools/xwiki-commons-tool-
verification-resources/src/main/java/org/xwiki/tool/checkstyle/
UnstableAnnotationCheck.java:[35,45]
 cannot find symbol
[ERROR]   symbol:   class AnnotationUtility
[ERROR]   location: package com.puppycrawl.tools.checkstyle.utils
[ERROR] /pipeline/source/.ci-temp/xwiki-commons/xwiki-commons-tools/xwiki-commons-tool-
verification-resources/src/main/java/org/xwiki/tool/checkstyle/
UnstableAnnotationCheck.java:[104,13]
 cannot find symbol
[ERROR]   symbol:   variable AnnotationUtility
[ERROR]   location: class org.xwiki.tool.checkstyle.UnstableAnnotationCheck
[ERROR] /pipeline/source/.ci-temp/xwiki-commons/xwiki-commons-tools/xwiki-commons-tool-
verification-resources/src/main/java/org/xwiki/tool/checkstyle/
UnstableAnnotationCheck.java:[105,32]
 cannot find symbol
[ERROR]   symbol:   variable AnnotationUtility
[ERROR]   location: class org.xwiki.tool.checkstyle.UnstableAnnotationCheck
[ERROR] -> [Help 1]
```

**Comment by romani** [comment]
@pbludov , do you plan to change package name `utils` to `util` in next PRs ?

if yes, that is good, it is better to be done in separate PR, to ease verification. But it will result in second commit of the same PR to xwiki - this is completely fine.

**Comment by pbludov** [comment]
> do you plan to change package name utils to util in next PRs ?

This sounds reasonable, but we also have the package names `checks`, `filters`, `grammars` etc.
Should they be renamed?

**Comment by pbludov** [comment]
> please confirm that you have such problem on your local, when you use binaries from your branch.

Yes, I was able to reproduce this error after cleaning the .m2/repositories folder.

**Comment by pbludov** [comment]
It seems they only need one method from the `AnnotaionUtility` class. We can keep this method for at least one release. In this case, there is no need to exclude `xwiki` from the build process.

**Comment by rnveach** [comment]
> we also have the package names checks, filters, grammars etc. Should they be renamed?

Not in this PR as the issue is only for utilities. If @romani agrees a new issue can be created for them.

> We can keep this method for at least one release. In this case, there is no need to exclude xwiki from the build process.

Re-read what @romani posted, we are not excluding them the build process and we will not keep a long method and class. We are doing something similar to https://github.com/checkstyle/checkstyle/commit/1695cabadb7b2e872df6083f0a56a4e35a5ecdd9#diff-ee38279d0500e28b395b2ead6f143baf
We are creating our own patch to their code and applying in the CI.

**Comment by romani** [comment]
>we also have the package names checks, filters, grammars etc. Should they be renamed?

Unfortunately we can not rename packages "checks" and "filters", they are in use in user configurations and plugins. It will be huge breaking compatibility. Names of all modules and their package names are kind of API.

But we can rename other packages, like "grammar". It is better to make it in separate PR or even issue.

>We can keep this method for at least one release

No need.
We need to do in our wercker config either patch apply or change clone url to your fork(where patch is present).
Please let me know if my comment still not clear.

All CIs should pass to let me merge this PR.

**Comment by pbludov** [comment]
>It seems they only need one method from the AnnotaionUtility class

I was wrong. The code reverted to https://github.com/Softus/checkstyle/commit/22aa43d4c4a37eee0ff76e078e2b86b9460ce8cc

File: wercker.yml [hunk 1]
```diff
@@ -68,16 +68,17 @@ build:
...
-      name: NoErrorTest - XWiki
-      code: |
-        if [[ $SKIP_CI == 'false' ]]; then
-          echo "Command: ./.ci/wercker.sh no-error-xwiki"
-          ./.ci/wercker.sh no-error-xwiki
-        else
-          echo "build is skipped ..."
-        fi
+# Temporary disabled as result of https://github.com/checkstyle/checkstyle/issues/5812
+# The check should be restored after the XWiki will be patched.
```
Comment (romani) [inline wercker.yml]: not sure why you disabled it completely
we benefit a lot using xwiki in regression testing.
Are you uncomfortable to make pull request to xwiki ? or just disabled to make all other fixes first ?
Just let me know I am eager to help.

Comment (rnveach) [reply wercker.yml]: See https://github.com/checkstyle/checkstyle/pull/5857#issuecomment-392784899 on example on how to create a patch

Comment (pbludov) [reply wercker.yml]: done

File: .ci/wercker.sh [hunk 2]
```diff
@@ -57,6 +57,10 @@ no-error-xwiki)
   checkout_from https://github.com/xwiki/xwiki-commons.git
   cd .ci-temp/xwiki-commons
   git checkout 44b0c0048c516dae20cf5f8a71181af836549484
+  # The classes AnnotationUtility and CommonUtils have been renamed
+  # https://github.com/checkstyle/checkstyle/issues/5812
```
Comment (romani) [inline .ci/wercker.sh]: this issue will be closed right after  merge.
please reference here link to PR to xwiki, to clearly define till what time hack should stay , comment should be like "till https:........"

File: wercker.yml [hunk 3]
```diff
@@ -68,7 +68,6 @@ build:
         else
           echo "build is skipped ..."
         fi
-
```
Comment (romani) [inline wercker.yml]: I do not see this "-" in master.

Comment (pbludov) [reply wercker.yml]: done

**Comment by rnveach** [comment]
@pbludov spellchecker is failing.

File: .ci/wercker.sh (hunk 2 above)
Comment (romani) [reply .ci/wercker.sh]: @pbludov , please look at my message in gitter.

**Comment by romani** [comment]
this is what was expected - https://github.com/checkstyle/checkstyle/pull/5887

**Pull request merged**